src/messages/textDocument_completion.cc

```diff
@@ -155,7 +155,8 @@ template <typename T> char *tofixedbase64(T input, char *out) {
 // when given 1000+ completion items.
 void FilterCandidates(Out_TextDocumentComplete *complete_response,
                       const std::string &complete_text, lsPosition begin_pos,
-                      lsPosition end_pos, bool has_open_paren) {
+                      lsPosition end_pos, bool has_open_paren,
+                      const std::string &buffer_line) {
```

Comment:
If you are happy to do more...

Since `buffer_line` is now passed as a parameter, `has_open_paren` can be removed and the `IsOpenParenOrAngle` logic can be inlined here (it can be simplified a bit).

Reply:
Done.

This feature is actually already in a broken state in master:

1. `auto [c, l] = position;` but position has `int line; int character;`.
2. It inserts `item.label` which is wrong if detailedLabel = true.